mystery-methods.rb

```diff
@@ -54,3 +74,11 @@ def mystery6(s)
     return mystery6(s[(space+1)..-1]) + " " + s[0...space]
   end
 end
+
+def mystery6_new(s)
+  if s == nil || s.length == 0
+    return ""
+  else
+    return s[-1] + mystery6_new(s[0...-1])
```

Comment:
This also could have been:

```
return mystery6(s[1..-1]) + s[0]
```

Which  does the same thing, but in other languages is more transferable (at least in Java). Both definitely work though, so whichever.